src/uu/env/src/env.rs

```diff
@@ -602,6 +673,33 @@ fn apply_specified_env_vars(opts: &Options<'_>) {
     }
 }
 
+fn apply_ignore_signal(opts: &Options<'_>) -> UResult<()> {
+    for &sig_value in &opts.ignore_signal {
+        let sig: Signal = (sig_value as i32)
+            .try_into()
+            .map_err(|e| std::io::Error::from_raw_os_error(e as i32))?;
+
+        ignore_signal(sig)?;
+    }
+    Ok(())
+}
+
+fn ignore_signal(sig: Signal) -> UResult<()> {
+    // SAFETY: this is unsafe because we're just ignoring the signal with the default signal handler SigIgn
+    let result = unsafe { signal(sig, SigIgn) }.map(|_| ());
```

Comment:
Hmm?

- It's fine to ignore the `Ok(_)` value, because we don't actually need it.
- The thing that needs to be justified is why this is "safe" in the sense of memory- and type-safety: https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html
- This is because the signal function is only safe under certain non-trivial conditions: https://docs.rs/nix/latest/nix/sys/signal/fn.signal.html
- Converting the value to `()` just to match on it seems redundant.

```suggestion
    // SAFETY: This is safe because we write the handler for each signal only once, and therefore "the current handler is the default", as the documentation requires it.
    let result = unsafe { signal(sig, SigIgn) };
```